src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-form/configuration-form.component.ts

```diff
@@ -0,0 +1,267 @@
+      { name: 'entity_addr_t', inputType: 'text', humanReadable: 'IPv4 or IPv6 address' },
+      { name: 'uuid_d', inputType: 'text', humanReadable: 'UUID' },
+      { name: 'bool', inputType: 'checkbox', humanReadable: 'Boolean value' }
+    ];
+
+    let currentType = null;
+
+    knownTypes.forEach(function(knownType) {
+      if (knownType.name === type) {
+        currentType = knownType;
+      }
+    });
+
+    if (currentType !== null) {
+      return currentType;
+    }
+
+    throw new Error('Found unknown type "' + type + '" for config option.');
+  }
+
+  getPatternHelpText(type: string): string {
+    const numberTypes = ['uint64_t', 'int64_t', 'size_t', 'secs'];
+
+    if (numberTypes.includes(type)) {
+      return 'The entered value needs to be a number.';
+    } else if (type === 'double') {
+      return 'The entered value needs to be a number or decimal.';
+    } else if (type === 'entity_addr_t') {
+      return 'The entered value needs to be a valid IP address.';
+    } else if (type === 'uuid_d') {
+      return (
+        'The entered value doesn\'t match the valid pattern ' +
+        '[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}'
+      );
+    }
+  }
+
+  getValidators(configOption: any): ValidatorFn[] {
+    const numberTypes = ['uint64_t', 'int64_t', 'size_t', 'double', 'secs'];
+
+    if (numberTypes.includes(configOption.type)) {
+      return this.numberValidator(configOption);
+    } else if (configOption.type === 'entity_addr_t') {
+      return [this.ipv4Ipv6Validator(configOption)];
+    } else if (configOption.type === 'uuid_d') {
+      return [this.uuidValidator(configOption)];
+    }
+  }
+
+  getStep(type: string, value: number): number | undefined {
+    const numberTypes = ['uint64_t', 'int64_t', 'size_t', 'secs'];
+
+    if (numberTypes.includes(type)) {
+      return 1;
+    }
+
+    if (type === 'double') {
+      if (value !== null) {
+        const stringVal = value.toString();
+        if (stringVal.indexOf('.') !== -1) {
+          // Value type double and contains decimal characters
+          const decimal = value.toString().split('.');
+          return Math.pow(10, -decimal[1].length);
+        }
+      }
+
+      return 0.1;
+    }
+
+    return undefined;
+  }
+
+  private numberValidator(configOption): ValidatorFn[] {
+    const validators = [];
+    if (configOption.max && configOption.max !== '') {
+      this.maxValue = configOption.max;
+      validators.push(Validators.max(configOption.max));
+    }
+    if (configOption.min && configOption.min !== '') {
+      this.minValue = configOption.min;
+      validators.push(Validators.min(configOption.min));
+    }
+    if (configOption.type === 'double') {
+      validators.push(Validators.pattern('^-?[0-9]+(.[0-9]+)?$'));
+    } else {
+      validators.push(Validators.pattern('^-?[0-9]+$'));
+    }
+
+    return validators;
+  }
+
+  private ipv4Ipv6Validator(configOption): ValidatorFn {
```

Comment:
Consider relocating this to shared/forms/cd-validators.ts. Think time will come that other forms will use this.